examples/stream-to-file/src/main.rs

```diff
@@ -84,6 +91,120 @@ async fn show_form() -> Html<&'static str> {
     )
 }
 
+// Handler that returns HTML for a multipart form.
+async fn show_form2() -> Html<&'static str> {
+    Html(
+        r#"
+        <!doctype html>
+        <html>
+            <head>
+                <title>Upload and Download!</title>
+            </head>
+            <body>
+                <h1>Upload and Download Files</h1>
+
+                <!-- Form for uploading files -->
+                <form action="/" method="post" enctype="multipart/form-data">
+                    <div>
+                        <label>
+                            Upload file:
+                            <input type="file" name="file" multiple>
+                        </label>
+                    </div>
+
+                    <div>
+                        <input type="submit" value="Upload files">
+                    </div>
+                </form>
+
+                <hr>
+                <!-- Buttons for downloading files -->
+                <form action="/file_download" method="get">
+                    <div>
+                        <input type="submit" value="Download file">
+                    </div>
+                </form>
+            </body>
+        </html>
+        "#,
+    )
+}
+
+/// A simpler file download handler that uses the `FileStream` response.
+/// Returns the entire file as a stream.
+async fn simpler_file_download_handler() -> Response {
+    //If you want to simply return a file as a stream
+    // you can use the from_path method directly, passing in the path of the file to construct a stream with a header and length.
+    FileStream::<AsyncReaderStream>::from_path("./CHANGELOG.md".into())
+        .await
+        .map_err(|_| (StatusCode::INTERNAL_SERVER_ERROR, "Failed to open file").into_response())
+        .into_response()
+}
+
+/// If you want to control the returned files in more detail you can implement a Stream
+/// For example, use the try_stream! macro to construct a file stream and set which parts are needed.
+async fn file_download_handler() -> Response {
+    let file_stream = match try_stream("./CHANGELOG.md", 5, 25, 10).await {
```

Comment:
`"./CHANGELOG.md"` will only works if we start the example from some directly, but I guess it's acceptable. We don't want to make the examples too complex.

Reply:
> `"./CHANGELOG.md"` will only works if we start the example from some directly, but I guess it's acceptable. We don't want to make the examples too complex.

Most of the time “. /CHANGELOG.md” exists, tried creating a file when the use example is started, it's too complicated for the example and may leave the file behind.

Reply:
What I mean is if I run `cd examples/stream-to-file && cargo run` then the path `"./CHANGELOG.md"` is not valid.

Reply:
> What I mean is if I run `cd examples/stream-to-file && cargo run` then the path `"./CHANGELOG.md"` is not valid.

Yes, perhaps it's better to create a file before launching the serve

Reply:
> What I mean is if I run `cd examples/stream-to-file && cargo run` then the path `"./CHANGELOG.md"` is not valid.

Now that I've modified the example, I'll create a file to test it first